drivers/serial/uart_nrfx_uarte.c

```diff
@@ -286,8 +820,24 @@ static int uarte_nrfx_poll_in(struct device *dev, unsigned char *c)
 static void uarte_nrfx_poll_out(struct device *dev,
 					 unsigned char c)
 {
-	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+	const struct uarte_nrfx_data *data = get_dev_data(dev);
 
+	if (data->async) {
+#if IS_ENABLED(CONFIG_UART_ASYNC_API)
+		uart_callback_t prev_cb = data->async->evt_handler;
+		void *usr_context = data->async->user_data;
+		volatile bool finished = false;
+
+		uarte_nrfx_callback_set(dev,
+					poll_out_callback,
+					(void *)&finished);
+		uarte_nrfx_tx(dev, &c, 1, 100);
+		while (!finished) {
+		}
+		uarte_nrfx_callback_set(dev, prev_cb, usr_context);
```

Comment:
This hack is really ugly. Especially with this hard-coded timeout. Please, use something nicer, for example, disable the `ENDTX` interrupt here or use a different handling of the `ENDTX` event in `uarte_nrfx_isr` if `data->async->tx` is `NULL`.